src/backend/base/langflow/components/prototypes/RunnableExecutor.py

```diff
@@ -108,11 +114,24 @@ def get_input_dict(self, runnable, input_key, input_value):
                 status = f"Warning: The input key is not '{input_key}'. The input key is '{runnable.input_keys}'."
         return input_dict, status
 
-    def build_executor(self) -> Message:
+    async def build_executor(self) -> Message:
         input_dict, status = self.get_input_dict(self.runnable, self.input_key, self.input_value)
-        result = self.runnable.invoke(input_dict)
+        if not isinstance(self.runnable, AgentExecutor):
+            raise ValueError("The runnable must be an AgentExecutor")
+
+        if self.use_stream:
+            return self.astream_events(input_dict)
+        else:
+            result = await self.runnable.ainvoke(input_dict)
         result_value, _status = self.get_output(result, self.input_key, self.output_key)
         status += _status
         status += f"\n\nOutput: {result_value}\n\nRaw Output: {result}"
         self.status = status
         return result_value
+
+    async def astream_events(self, input):
+        async for event in self.runnable.astream_events(input, version="v1"):
```

Comment:
I think what you could do is use the callbacks's methods for each type of event.